### PR TITLE
[Datasets] Add basic data ecosystem overview, user guide links, other data processing options card.

### LINKED ISSUE #23089: [Datasets] [Docs] Create data ecosystem overview
The Ray Data landing page should contain a brief overview of the data ecosystem, positioning Ray Datasets as the Ray-native offering.

### PR DESCRIPTION
**Pull request by clarkzinzow**
Tweaks the landing page for Datasets to give a (very brief) overview of the data ecosystem. Also includes misc. additions, e.g. to the new user guide pages.

Expanding the data ecosystem overview to include more content can be done in a future PR.

Closes #23089

### CODE REVIEW

File: doc/source/data/dataset.rst [hunk 1]
```diff
@@ -3,9 +3,22 @@
 .. _datasets:
 
 ==================================================
-Ray Datasets: Distributed Data Loading and Compute
+Ray Data: Data Procesing on Ray
```
Comment (ericl) [inline doc/source/data/dataset.rst:6]:
```suggestion
Ray Data: Data Processing on Ray
```

File: doc/source/data/dataset.rst [hunk 2]
```diff
@@ -3,9 +3,22 @@
...
 ==================================================
 
+The Ray ecosystem offers several great solutions for performant, large-scale data processing on Ray.
```
Comment (ericl) [inline doc/source/data/dataset.rst:9]:
```suggestion
The Ray ecosystem offers several solutions for performant, large-scale data processing on Ray.
```

File: doc/source/data/dataset.rst [hunk 3]
```diff
@@ -3,9 +3,22 @@
...
+
+:ref:`Ray Datasets<datasets-intro>` is Ray's first-class Ray-native offering, targeting large-scale data ingest
+for distributed training, scalable and resource-efficient batch inference, and performant last-mile preprocessing for both training and inference pipelines. Datasets is a great post-ETL data processing solution that strongly integrates with the rest of Ray's ML ecosystem.
+
+For running super scalable ETL pipelines, check out :ref:`Spark-on-Ray<spark-on-ray>`. For scaling up your data science workloads, check out :ref:`Dask-on-Ray<dask-on-ray>`, :ref:`Modin<modin`, and :ref:`Mars-on-Ray<mars-on-ray>`.
```
Comment (ericl) [inline doc/source/data/dataset.rst:14]:
```suggestion
For running ETL pipelines, check out :ref:`Spark-on-Ray<spark-on-ray>`. For scaling up your data science workloads, check out :ref:`Dask-on-Ray<dask-on-ray>`, :ref:`Modin<modin`, and :ref:`Mars-on-Ray<mars-on-ray>`.
```

**Review by ericl — CHANGES_REQUESTED** [review]
After thinking about this more, I don't think this change is desirable. Some of these integrations are not quite fully baked, so we shouldn't be recommending them in the front page.

Saying there are "several solutions" is also confusing the messaging here. We want to be clear that Datasets is the one standard way, and integrations can be introduced later.

For this PR, I think the changes to the front matter of dataset.rst should be reverted, but we can keep the changes further down in the page.

**Comment by clarkzinzow** [comment]
Not stale

**Comment by jianoaix** [comment]
> After thinking about this more, I don't think this change is desirable. Some of these integrations are not quite fully baked, so we shouldn't be recommending them in the front page.
> 
> Saying there are "several solutions" is also confusing the messaging here. We want to be clear that Datasets is the one standard way, and integrations can be introduced later.
> 
> For this PR, I think the changes to the front matter of dataset.rst should be reverted, but we can keep the changes further down in the page.

+1 not to feature alternative solutions at home page.
I think it's a good idea to having an overview of where Dataset fits / positioning, e.g. one reads this page should easily know that this is not for ETL.

**Comment by ericl** [comment]
Stale.

**Comment by clarkzinzow** [comment]
I'll be revisiting this PR today and implementing the above feedback.

**Comment by clarkzinzow** [comment]
@ericl I implemented all feedback, including pushing all mentioning of third-party integrations to the integrations card. Can you and @jianoaix PTAL?

**Review by ericl — COMMENTED** [review]
I feel very strongly that we shouldn't introduce "Ray Data" or "broader ecosystem" in the frontmatter and the doc should focus purely on Datasets. I made a commit to revert this change.

If it looks good to you, the rest of the PR is fine with me.

**Comment by clarkzinzow** [comment]
@ericl Sure, if that commit makes this pass the bar, can you approve the PR so I can merge it?

**Review by ericl — APPROVED** [review]

**Pull request merged**